api.go

```diff
@@ -0,0 +1,56 @@
+package sqlparser
+
+// Visit defines the signature of a function that
+// can be used to visit all nodes of a parse tree.
+type Visit func(node Node) (kontinue bool, err error)
```

Comment:
Surfacing note: So technically is "fine" to mutate the `node` in the `Visit(...)` function. Looking at how this works internally, the walking logic will account for whatever mutation you did since we first call `Visit(...)` and then we walk  subtree. That order implies that whatever you did in the `node`, will direct the subtree walk.

By "mutate" I refer editing internal elements. The client can't actually change the `node` type.

Makes sense to you?

Reply:
As a tangent, I was also wondering if we could use this to replace future `TXN_HASH()` calls with concrete values, but I don't think that's possible. (It's fine; just saying it makes sense to you too). 

The logic would be similar regarding walking, but probably the API to do the node replacement will be a bit different.

Reply:
yeah makes sense. @carsonfarmer is planning to use this to mutate the table's name. Do you see any risk here? The client can probably mess up the walker by mutating things, but it probably not make sense to do it.

We could probably use it to replace `TXN_HASH()`. Gotta think a bit about it. However, if I remember correctly, those functions are used in `INSERT` or `UPDATE` queries which are much simpler, so it is probably easier to just create a replace method in those statements.

Reply:
Changing simple attributes in nodes should probably be fine (e.g: table names).  I mean, I think it's safe in general because of how walkSubree is called after calling `Visit` (thus after the mutation), so what will happen in the next steps of the walk will be coherent with the mutation.

Reply:
Excellent!